src/main/java/com/iota/iri/network/Node.java

```diff
@@ -504,6 +515,49 @@ private Runnable spawnBroadcasterThread() {
         };
     }
 
+    /**
+     * This method returns the lambda for the requester {@link Thread} that tries to actively request missing
+     * transactions from the request queue by sending random tips together with our request.
+     *
+     * Without this thread we are limited to the amount of incoming transactions while syncing the node, because we can
+     * only request new transactions when we are re-broadcasting these received transactions.
+     *
+     * To not send unnecessary requests for "temporarily" missing transactions while receiving broadcasts from the
+     * network, we only trigger its logic when the queue exceeds the {@link #REQUESTER_THREAD_ACTIVATION_THRESHOLD}.
+     *
+     * @return lambda for the requester {@link Thread}
+     */
+    private Runnable spawnRequesterThread() {
```

Comment:
Maybe open an issue about doing something a bit more efficient.
Instead of having an infinite while loop maybe you can initiate this thread once the threshold has been exceeded.